peakina/readers/excel.py

```diff
@@ -0,0 +1,132 @@
+"""
+Module to add xml support
+"""
+import enum
+from io import StringIO
+from typing import Any, Dict, List
+
+import pandas as pd
+import xlrd
+from openpyxl import load_workbook
+from openpyxl.utils.exceptions import InvalidFileException
+
+
+class EXCEL_TYPE(enum.Enum):
+    NEW = "new"
+    OLD = "old"
+
+
+def _yielder(preview: Dict[str, int], sheet_name: Any) -> Any:
+    """
+    A generator for old excel types
+    """
+
+    if bool(preview):
+        limit, offset = preview.get("nrows", 1), preview.get("offset", 0)
+        to_iter = range(offset, offset + limit + 1)
+    else:
+        to_iter = range(sheet_name.nrows)
+
+    for rx in to_iter:
+        yield sheet_name.row(rx)
+
+
+def _read_old_xls_format(wb: Any, sh_name: str, preview: Dict[str, int]) -> Any:
+    """ """
+    return list(_yielder(preview, wb[sh_name]))
```

Comment:
Probably better to put everything here. You use `_yielder` only once. So the lazy generation is useless here

Reply:
👍🏾 Updated !